WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -1003,27 +1019,27 @@ private boolean shouldShowListPreference(DetailListPreference preference) {
     private String[] createLanguageDisplayStrings(CharSequence[] languageCodes) {
         if (languageCodes == null || languageCodes.length < 1) return null;
 
-        String[] displayStrings = new String[languageCodes.length];
-
+        Locale deviceLocale = WPPrefUtils.languageLocale(null);
+        String[] entryStrings = new String[languageCodes.length];
         for (int i = 0; i < languageCodes.length; ++i) {
-            displayStrings[i] = StringUtils.capitalize(getLanguageString(
-                    String.valueOf(languageCodes[i]), WPPrefUtils.languageLocale(languageCodes[i].toString())));
+            entryStrings[i] = StringUtils.capitalize(
+                    getLanguageString(languageCodes[i].toString(), deviceLocale)) + "__" + languageCodes[i];
```

Comment:
@tonyr59h: I know this is already merged, but why do we need the extra "__" bit? I moved these 3 methods to `WPPrefUtils` in my branch to reuse in Account Settings and this feels pretty hacky to me. It doesn't really return the display strings either. Can we maybe omit the "__" bit and set the entry values from the language array instead in `sortLanguages()`?